### PR TITLE
Port planck's eval implementation

### PR DESCRIPTION
**Pull request by moxaj**
See #146

Question: should the new `lumo.repl/eval` be exposed anywhere else?

### CODE REVIEW

File: src/cljs/snapshot/lumo/repl.cljs [hunk 1]
```diff
@@ -806,6 +807,57 @@
                            (filter matches? (public-syms ns)))))
                   (all-ns)))))
 
+;; Taken from planck eval implementation
+;; The following atoms and fns set up a scheme to
+;; emit function values into JavaScript as numeric
+;; references that are looked up.
+
+(defonce ^:private fn-index (atom 0))
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: these don't need to be atoms. can you switch them to volatiles?

File: src/cljs/snapshot/lumo/repl.cljs [hunk 2]
```diff
@@ -806,6 +807,57 @@
...
+(defonce ^:private fn-refs (atom {}))
+
+(defn- clear-fns!
+  "Clears saved functions."
+  []
+  (reset! fn-refs {}))
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: `vreset!` once they're volaties

File: src/cljs/snapshot/lumo/repl.cljs [hunk 3]
```diff
@@ -806,6 +807,57 @@
...
+
+(defn- put-fn
+  "Saves a function, returning a numeric representation."
+  [f]
+  (let [n (swap! fn-index inc)]
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: `vswap!`

**Comment by anmonteiro** [comment]
This looks great! Left a few inline comments and I'll merge once you address them.

Thanks!

File: src/cljs/snapshot/lumo/repl.cljs (hunk 1 above)
Comment (mfikes) [inline src/cljs/snapshot/lumo/repl.cljs]: Great ideas are always worth stealing. mfikes/planck@0f29414

**Comment by moxaj** [comment]
Sorry about the messy history! The windows specific commit should go in another branch.

**Comment by anmonteiro** [comment]
thanks!

**Pull request merged**